### DISCUSSION TITLE
Question about the the demo

### DISCUSSION THREAD

**Discussion by chao0x11** [openssl/openssl#26086] [Q&A]
I'm learning the demo in [openssl-3.4.0/demos/pkey/EVP_PKEY_EC_keygen.c](https://fossies.org/linux/openssl/demos/pkey/EVP_PKEY_EC_keygen.c), and then I use DER-encoder to encode my EC keys. The size of the result key is 118 (passphrase != NULL) and 88 (passphrase == NULL). While in the demo [EVP_EC_Signature_demo.h](https://fossies.org/linux/openssl/demos/signature/EVP_EC_Signature_demo.h) the key length is much larger than I got. How can I get the same length of keys in [EVP_EC_Signature_demo.h](https://fossies.org/linux/openssl/demos/signature/EVP_EC_Signature_demo.h)?
p.s., I use secp256k1 in my test.

**Comment by levitte** [comment]
Would you mind showing us the encoding code?

**Reply by chao0x11** [reply]
The encoder and decoder file: https://fossies.org/linux/openssl/demos/encode/ec_encode.c
